test.cpp

```diff
@@ -14,6 +15,93 @@ using namespace autograd;
 #define CUDA_GUARD std::cerr << "No cuda, skipping test" << std::endl; return
 #endif
 
+class CartPole {
```

Comment:
Is this derived from some existing example? If yes, maybe add a short comment pointing to the source.

Reply:
Comment was in the step function, but I moved it up here.